src/containers/header/Header.jsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+
+import Logo from '../../components/common/Logo';
+import Button from '../../components/common/Button';
+import SearchForm from './search-form/SearchForm';
+import ResultsBanner from './results-banner/ResultsBanner';
+import MovieDetails from './movie-details/MovieDetails';
+
+import moviesMock from '../../movies.mock.json';
+
+import './header.scss';
+
+export default class Header extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            searchText: '',
+            searchBy: 'title',
+            sortBy: 'release',
+            resultsQuantity: undefined
+        };
+    }
+
+    handleSearchBarChange = (event) => {
+        const { value } = event.target;
+
+        console.log('Search value: ', value);
+
+        this.setState({
+            searchText: value
+        });
+    }
+
+    handleSearchReturnClick = () => {
+        console.log('Return to search form');
+    }
+
+    handleSearchByChange = (type) => {
+        this.setState({
+            searchBy: type
+        });
+    }
+
+    handleSearchButtonClick = (event) => {
+        console.log('Searching ...');
+
+        const { data } = moviesMock;
+
+        this.setState({
+            resultsQuantity: data.length
+        });
+
+        this.props.onSearchResultsFetch(data);
+    }
+
+    handleSortByChange = (type) => {
+        this.setState({
+            sortBy: type
+        });
+    }
+
+    render() {
+        console.log('rendering Header ...');
+
+        const { movieData } = this.props;
+        const { resultsQuantity } = this.state;
+
+        const content = !movieData ?
+            <SearchForm
+                onSearchBarChange={ this.handleSearchBarChange }
```

Comment:
I was suggested by an article:
https://reactjs.org/docs/lifting-state-up.html

I just wanted to have all state in common parent component which is `Header` in my case to communicate with siblings components of `SearchForm` like `ResultsBanner` and to pass search results to the upper component which is `Root` to communicate with `MoviesList` component.

Probably this is not a good idea, because of redundant re-rendering of unrelated components, but how to pass search results to the `MoviesList` which is outside of `Header` component?